content/en/docs/tasks/tls/manual-rotation-of-ca-certificates.md

```diff
@@ -0,0 +1,154 @@
+For more information about best practices for CA certificates, see [Single root CA](docs/setup/best-practices/certificates/#single-root-ca).
+
+{{% /capture %}}
+
+{{% capture steps %}}
+## Choose an approach
+There are two ways to rotate or replace CA certificates.
+
+## Replace the cluster (recommended) {#replace-the-cluster}
+
+This is the safest approach to rotate or replace the CA certificates.
+
+1. Create a new cluster with new CA certificates.
+
+2. Migrate the workloads to the new cluster.
+
+3. Remove or destroy the old cluster.
+
+## Rotate the CA certificates manually
+
+{{< caution >}}
+Make sure to back up your certificate directory along with configuration files (.conf) and any other necessary files.
+
+Manually rotating the CA works by timing when a Kubernetes cluster knows about new certificates. Do not restart any pods or update any component in the cluster while manually rotating the CA.
+
+This approach requires a considerable downtime of your cluster.
+
+{{< /caution >}}
+
+1. Back up your `/etc/kubernetes` directory, referred to as the *Kubernetes conf directory* and `/etc/kubernetes/pki`, referred to as the *Kubernetes certificates directory*.
+    
+1. Distribute new CA private keys (`ca.key` and `front-proxy-ca.key`) to all your control plane nodes in the Kubernetes certificates directory.
+
+1. Append new CA to the existing CA files (`ca.crt` and `front-proxy-ca.crt`) and the file against `--client-ca-file` flag in the `kube-apiserver` and `kube-scheduler` manifest files.
+
+    {{< note >}}
+
+    If you used `kubeadm` to deploy your cluster, the control plane components run as {{< glossary_tooltip text="static Pods" term_id="static-pod" >}}, managed by the {{< glossary_tooltip text="kubelet" term_id="kubelet" >}}. Components are not disrupted by in-place updates of CA certificates until someone kills or restarts an already running Pod.
+
+    {{< /note >}}
+
+1. Update *Kubernetes controller manager's* `--root-ca-file` to include both old and new CA.
+
+     {{< note >}}
+
+     Remove the flag `--client-ca-file` from *Kubernetes controller manager* manifest file or replace the file with new CA. [Issue 1350](https://github.com/kubernetes/kubeadm/issues/1350) enumerates an issue with *Kubernetes controller manager* being unable to accept CA bundle.
+
+    {{< /note >}}
+
+1. Copy the new CA certificate file on all nodes needed by the kubelet.
+
+1. Get new certificates and private keys signed from the new CA for `apiserver.crt`, `apiserver-kubelet-client.crt`, and `front-proxy-client.crt` and replace them in the Kubernetes certificates directory.
+
+    To generate certificates and private keys for your cluster using the `openssl` command line tool, see [Certificates (`openssl`)](/docs/concepts/cluster-administration/certificates/#openssl).
+    
+    You can also use [`cfssl`](/docs/concepts/cluster-administration/certificates/#cfssl).
```

Comment:
_Optional tweak_
The reader needs to use a third party tool to achieve this outcome. I'd also let them know about smallstep:
```suggestion
    You can also use other tools such as [`cfssl`](/docs/concepts/cluster-administration/certificates/#cfssl) or [`step`](https://smallstep.com/cli/).
```

Reply:
I will ignore this one we have given 2 options already openssl and cfssl.